modules/layers/src/layers/editable-h3-layer.ts

```diff
@@ -0,0 +1,152 @@
+import EditableLayer from './editable-layer';
+
+const DEFAULT_EDIT_MODE = ViewMode;
+const DEFAULT_H3_RESOLUTION = 9;
+
+const defaultProps = {
+  mode: DEFAULT_EDIT_MODE,
+
+  // EditableGeoJsonLayer
+  geoJsonData: {},
+  ...EditableGeoJsonLayer.defaultProps,
+
+  // h3 layer
+  h3Data: {},
+  filled: false,
+  stroked: true,
+  lineWidthScale: 1,
+  lineWidthMinPixels: 1,
+  lineWidthMaxPixels: Number.MAX_SAFE_INTEGER,
+  lineWidthUnits: 'pixels',
+  getHexagons: (d) => d.hexIds,
+  resolution: DEFAULT_H3_RESOLUTION,
+};
+
+export default class EditableH3Layer extends EditableLayer {
+  static layerName = 'EditableH3Layer';
+  static defaultProps = defaultProps;
+
+  initializeState() {
+    super.initializeState();
+
+    this.setState({
+      tentativeHexagonIDs: [],
+    });
+  }
+
+  getDerivedHexagonIDs(coords) {
+    return polyfill(coords, this.props.resolution, true);
+  }
+
+  renderLayers() {
+    const layers: any = [
+      new EditableGeoJsonLayer(
+        this.getSubLayerProps({
+          id: 'editable-geojson',
+
+          mode: this.props.mode,
+          data: {
+            type: 'FeatureCollection',
+            features: [],
+          },
+          selectedFeatureIndexes: this.props.selectedFeatureIndexes || [],
+
+          onEdit: (editAction) => {
+            const { editType, editContext } = editAction;
+
+            switch (editType) {
+              case 'updateTentativeFeature':
+                // tentative feature updates, updated on every pointer move
+                if (editContext.feature.geometry.type === 'Polygon') {
+                  const coords = editContext.feature.geometry.coordinates;
+                  const hexIDs = this.getDerivedHexagonIDs(coords);
+
+                  this.setState({ tentativeHexagonIDs: hexIDs });
+                }
+                break;
+              case 'addFeature':
+                // once the shape is finished drawing, calculate final hexagons
+                let finalHexagonIDs;
+                const committedHexagonIDs = new Set(this.getHexIdsFromH3Data());
+                const tentativeHexagonIDs = new Set(this.state.tentativeHexagonIDs);
+
+                switch (this.props.modeConfig.booleanOperation) {
+                  case 'union':
+                    finalHexagonIDs = [
+                      ...new Set([...committedHexagonIDs, ...tentativeHexagonIDs]),
+                    ];
+                    break;
+                  case 'intersection':
+                    finalHexagonIDs = [...committedHexagonIDs].filter((hexID: string) =>
+                      tentativeHexagonIDs.has(hexID)
+                    );
+                    break;
+                  case 'difference':
+                    finalHexagonIDs = [...committedHexagonIDs].filter(
+                      (hexID: string) => !tentativeHexagonIDs.has(hexID)
+                    );
+                    break;
+                  default:
+                    break;
```

Comment:
perhaps we should union by default rather than doing nothing (after the user spent their time drawing a polygon).
```
switch(...) {
  case 'union':
  default:
    // do the union logic
  case 'intersection':
    ...
}
```